### PR TITLE
Add GitHub Workflows for vtadmin-web CI

### PR DESCRIPTION
**Pull request by doeg**
## Description

🎃 **Please squash-merge this PR.** I left a verbose commit history since the failing tests prove it's working. 😎 

This adds a new GitHub Workflow, [vtadmin_web_ci](https://github.com/vitessio/vitess/actions/workflows/vtadmin_web_ci.yml), which is run on pull requests that contain changes in the `web/vtadmin` directory. 

Currently, it does the following in sequence:
- Ensure vtadmin-web builds
- Run the unit tests
- Run the linters

As noted, more parallelization will make a decent future enhacement... but iterating on Actions is slooowwww (~20 min of queue time) and this new workflow is already pretty fast (~2 min runtime) with minimal effort/complexity.

## Related Issue(s)

N/A

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

## Deployment Notes

N/A

## Impacted Areas in Vitess
Components that this PR will affect:

- [ ]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build/CI
- [x]  VTAdmin

### CODE REVIEW

File: web/vtadmin/src/api/http.test.ts [hunk 1]
```diff
@@ -17,7 +17,6 @@ import { rest } from 'msw';
 import { setupServer } from 'msw/node';
 
 import * as api from './http';
-import { vtadmin as pb } from '../proto/vtadmin';
```
Comment (ajm188) [inline web/vtadmin/src/api/http.test.ts:20]: Was this removed just because it was unused?

**Review by ajm188 — APPROVED** [review]
this is awesome!!

File: web/vtadmin/src/api/http.test.ts (hunk 1 above)
Comment (doeg) [reply web/vtadmin/src/api/http.test.ts:20]: Yeah, it actually added a nice annotation + everything:

Screen Shot 2021-03-08 at 9 22 39 PM

Comment (ajm188) [reply web/vtadmin/src/api/http.test.ts:20]: love to see it, sincerely

**Review by rohit-nayak-ps — CHANGES_REQUESTED** [review]
We now have a generator script: `test/ci_workflow_gen.go` that creates CI workflow files for easier maintenance. Can you please modify that script and run it to generate your new test. We currently have two types of tests: unit and cluster tests, each with its own template. This will be a new type of workflow since its template is different from these, so you will need to update the script appropriately. Thanks!

`make generate_ci_workflows` generates these workflows.

**Comment by doeg** [comment]
@rohit-nayak-ps thanks for the pointer! I can see how that will be useful.

I have something in mind for this but I'm not going to get to it this week, since we're getting ready for the Thursday demo. :) I'm going to close this PR -- it's useful for archival purposes, but it's going to accrue _so_ many conflicts until I get around to it, haha.

**Pull request closed without merging**